src/code/z_play_hireso.c

```diff
@@ -1,27 +1,1349 @@
+        rectLeft += 150;
+    }
+
+    gDPPipeSync(gfx++);
+    *gfxP = gfx;
+}
+
+s16 sBombersNotebookColumnColors[][2][3] = {
+    { { 165, 183, 195 }, { 140, 158, 170 } },
+    { { 197, 195, 172 }, { 172, 170, 147 } },
+    { { 223, 190, 200 }, { 190, 165, 175 } },
+};
+
+void BombersNotebook_DrawColumns(Gfx** gfxP) {
+    Gfx* gfx = *gfxP;
+    s16* color;
+    s32 i;
+    s32 j;
+    s32 k;
+    s32 columnRectLeft;
+    s32 subColumnRectLeft;
+    s32 pad;
+
+    gDPLoadTextureBlock_4b(gfx++, gBombersNotebookSubColumnBoxTex, G_IM_FMT_I, 48, 1, 0, G_TX_NOMIRROR | G_TX_WRAP,
+                           G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD);
+
+    color = &sBombersNotebookColumnColors[0][0][0];
+    for (i = 0, columnRectLeft = 120; i < ARRAY_COUNT(sBombersNotebookColumnColors); i++, columnRectLeft += 150) {
+        subColumnRectLeft = columnRectLeft;
+        for (j = 0; j < 2; j++) {
+            gDPSetPrimColor(gfx++, 0, 0, color[0], color[1], color[2], 192);
+            for (k = 0; k < 2; k++) {
+                BombersNotebook_TextureRectangle(&gfx, subColumnRectLeft * 4, 104 * 4, (subColumnRectLeft + 48) * 4,
+                                                 480 * 4, 0, 0, 0, 0x400, 0x400);
+                subColumnRectLeft += 36;
+            }
+            color += 3;
+        }
+    }
+
+    gDPPipeSync(gfx++);
+    *gfxP = gfx;
+}
+
+TexturePtr sBombersNotebookEventIconTextures[] = {
+    gBombersNotebookEntryIconExclamationPointTex,
+    &D_0700AC00,
+    &D_0700AEA0,
+};
+s16 sBombersNotebookEntryIconColors[][3] = {
+    { 255, 255, 0 },
+    { 141, 255, 182 },
+    { 255, 100, 60 },
+};
+
+void BombersNotebook_DrawEntries(Gfx** gfxP, s32 row, u32 rectTop) {
+    Gfx* gfx = *gfxP;
+    s32 entryRectLeft;
+    u32 eventIcon;
+    u32 unfinishedEvent = false;
+    u16 startTime;
+    u16 endTime;
+    s32 j = 0;
+    u32 iconRectLeft;
+    s32 entryRectRight;
+    u32 yOffset;
+    s32 entryWidth;
+
+    while (true) {
+        if (sBombersNotebookEntries[row][j] == BOMBERS_NOTEBOOK_ENTRY_END) {
+            if (!unfinishedEvent) {
+                gDPPipeSync(gfx++);
+                gDPSetCombineLERP(gfx++, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0,
+                                  PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0);
+                gDPSetPrimColor(gfx++, 0, 0, 0, 0, 0, 255);
+                gDPSetEnvColor(gfx++, 0, 0, 0, 255);
+                gDPLoadTextureBlock(gfx++, gBombersNotebookStampTex, G_IM_FMT_IA, G_IM_SIZ_8b, 32, 48, 0,
+                                    G_TX_MIRROR | G_TX_WRAP, G_TX_MIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                                    G_TX_NOLOD, G_TX_NOLOD);
+                BombersNotebook_TextureRectangle(&gfx, 567 * 4, (rectTop + 1) * 4, 599 * 4, (rectTop + 49) * 4, 0, 0, 0,
+                                                 0x400, 0x400);
```

Comment:
`0x400` can be `1 << 10` in many places. Also, I think `567 * 4` to `567 << 2` should be fine for constants, and leave `* 4` only when necessary for matching, which I acknowledge is in quite a few places. I think that's what @Thar0 prefers after dealing with this a lot in z_parameter

Reply:
I don't mind doing `1 << 10` for the `dsdx/dtdy` terms, but since `*4` is used basically everywhere I would personally prefer just to use `*4` for those constants as well.

Reply:
I'm not too picky about it. But I think the `1 << 10` should be applied at least